extensions/libarchive/MergeContent.h

```diff
@@ -259,20 +259,56 @@ class ZipMerge: public ArchiveMerge, public MergeBin {
   }
 };
 
+class AttributeMerger {
+public:
+  explicit AttributeMerger(std::deque<std::shared_ptr<org::apache::nifi::minifi::core::FlowFile>> &flows)
+    : flows_(flows) {}
+  void mergeAttributes(core::ProcessSession *session, std::shared_ptr<core::FlowFile> &merge_flow);
+  virtual ~AttributeMerger() = default;
+protected:
+  std::map<std::string, std::string> getMergedAttributes();
+  virtual void processFlowFile(const std::shared_ptr<core::FlowFile> &flow_file, std::map<std::string, std::string>& merged_attributes) = 0;
+
+  const std::deque<std::shared_ptr<core::FlowFile>> &flows_;
+};
+
+class KeepOnlyCommonAttributesMerger: public AttributeMerger {
+public:
+  explicit KeepOnlyCommonAttributesMerger(std::deque<std::shared_ptr<org::apache::nifi::minifi::core::FlowFile>> &flows)
+    : AttributeMerger(flows) {}
+protected:
+  void processFlowFile(const std::shared_ptr<core::FlowFile> &flow_file, std::map<std::string, std::string>& merged_attributes) override;
+};
+
+class KeepAllUniqueAttributesMerger: public AttributeMerger {
+public:
+  explicit KeepAllUniqueAttributesMerger(std::deque<std::shared_ptr<org::apache::nifi::minifi::core::FlowFile>> &flows)
+    : AttributeMerger(flows) {}
+protected:
+  void processFlowFile(const std::shared_ptr<core::FlowFile> &flow_file, std::map<std::string, std::string>& merged_attributes) override;
+
+private:
+  std::vector<std::string> removed_attributes_;
+};
+
 // MergeContent Class
 class MergeContent : public processors::BinFiles {
- public:
+public:
+  static constexpr const char *ATTRIBUTE_STRATEGY_KEEP_COMMON = "Keep Only Common Attributes";
```

Comment:
I'm all for using static members to store the attribute name, but then we should make all of them static members not just the new ones, if moving the others seems problematic, we should reassess leaving them be for now

Reply:
Instead of members, I would move them to the implementation file into a private namespace if possible.

Reply:
It's possible, but it would also require moving the ArchiveMerge::WriteCallback's and the MergeContent's constructor's implementation to the cpp file. In that case it might be better to move all the implementation details to the cpp file.

Reply:
I left them as ::processor namespace level constants as both tests and different classes use these constants currently.